app/Models/Beatmapset.php

```diff
@@ -544,6 +566,26 @@ public function nominate(User $user)
         ];
     }
 
+    public function love(User $user)
+    {
+        if (!$this->isLoveable()) {
+            return [
+                'result' => false,
+                'message' => trans('beatmaps.nominations.incorrect_state'),
+            ];
+        }
+
+        $this->getConnection()->transaction(function () use ($user) {
```

Comment:
I think this transaction should include the same tasks as qualifying a beatmap? (delete user ratings, check covers, delete scores, make a global event)

Reply:
loving skips most of that apparently

Reply:
well, it skips all of it atm because maps are only loved manually. seems like it should do all of that though?